lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -327,6 +323,75 @@ void setWantConsolidation(MHWRender::MRenderItem& renderItem, bool state)
     renderItem.setWantSubSceneConsolidation(state);
 #endif
 }
+
+PrimvarInfo* _getInfo(const PrimvarInfoMap& infoMap, const TfToken& token)
+{
+    auto it = infoMap.find(token);
+    if (it != infoMap.end()) {
+        return it->second.get();
+    }
+    return nullptr;
+}
+
+void _getColorData(
+    PrimvarInfoMap&  infoMap,
+    VtVec3fArray&    colorArray,
+    HdInterpolation& interpolation)
+{
+    PrimvarInfo* info = _getInfo(infoMap, HdTokens->displayColor);
+    if (info) {
+        const VtValue& value = info->_source.data;
+        if (value.IsHolding<VtVec3fArray>() && value.GetArraySize() > 0) {
+            colorArray = value.UncheckedGet<VtVec3fArray>();
+            interpolation = info->_source.interpolation;
+        }
+    } else {
+        // If color/opacity is not found, the 18% gray color will be used
+        // to match the default color of Hydra Storm.
+        TF_VERIFY(colorArray.empty());
+        colorArray.push_back(GfVec3f(0.18f, 0.18f, 0.18f));
+        interpolation = HdInterpolationConstant;
+
+        infoMap[HdTokens->displayColor] = std::make_unique<PrimvarInfo>(
+            PrimvarSource(VtValue(colorArray), interpolation, PrimvarSource::CPUCompute), nullptr);
+    }
+}
+
+void _getOpacityData(
+    PrimvarInfoMap&  infoMap,
+    VtFloatArray&    opacityArray,
+    HdInterpolation& interpolation)
+{
+    PrimvarInfo* info = _getInfo(infoMap, HdTokens->displayOpacity);
+    if (info) {
+        const VtValue& value = info->_source.data;
+        if (value.IsHolding<VtFloatArray>() && value.GetArraySize() > 0) {
+            opacityArray = value.UncheckedGet<VtFloatArray>();
+            interpolation = info->_source.interpolation;
+        }
+    } else {
+        TF_VERIFY(opacityArray.empty());
+        opacityArray.push_back(1.0f);
+        interpolation = HdInterpolationConstant;
+
+        infoMap[HdTokens->displayOpacity] = std::make_unique<PrimvarInfo>(
+            PrimvarSource(VtValue(opacityArray), interpolation, PrimvarSource::CPUCompute),
+            nullptr);
+    }
+}
+
+//! Access the points
+VtVec3fArray _points(PrimvarInfoMap& infoMap)
+{
+    if (PrimvarInfo* info = _getInfo(infoMap, HdTokens->points)) {
```

Comment:
Looks like we never expect the points to be unavailable. Maybe we should add a TF_VERIFY here.

Reply:
If someone calls _points and they don't exist in the infoMap they'll get back an empty VtVec3fArray.